### DISCUSSION TITLE
Current_NotificationActionTapped doesn't work, need help

### DISCUSSION THREAD

**Discussion by Cinmay2014** [thudugala/Plugin.LocalNotification#394] [Q&A] [answered]
Hi, need help.
I read Issues and discussions related to NotificationTapped to modified my code and seems everything is correct, but the current_NotificationActionTapped() method never been hit.
As you can see I add some debug lines in my code, in MainActivity I tried to check if OnNewIntent() work or not and the output showed it worked fine 
Need help.

App.xaml.cs
    public partial class App
    {
     
     
        public App(IPlatformInitializer initializer) : base(initializer)
        {
            InitializeComponent();
            LocalNotificationCenter.Current.NotificationActionTapped += Current_NotificationActionTapped;
            System.Diagnostics.Debug.WriteLine("Notification created");
        }

        

        protected override async void OnInitialized()
        {

            await NavigationService.NavigateAsync("NavigationPage/MainPage");
           
        }

        private void Current_NotificationActionTapped(NotificationActionEventArgs e)
        {
            Console.WriteLine("Notification tapped!");
            System.Diagnostics.Debug.WriteLine("Notification tapped!");
            Device.BeginInvokeOnMainThread(async () => {
                await App.Current.MainPage.DisplayAlert("Notification Tapped!", e.Request.ReturningData, "Done");
            });
        }

        protected override void RegisterTypes(IContainerRegistry containerRegistry)
        {
            containerRegistry.RegisterSingleton<IWebAuthenticator, WebAuthenticatorImplementation>();
            

            containerRegistry.RegisterForNavigation<NavigationPage>();
            containerRegistry.RegisterForNavigation<MainPage, MainPageViewModel>();
            containerRegistry.RegisterForNavigation<ViewA, ViewAViewModel>();
            containerRegistry.RegisterForNavigation<ViewB, ViewBViewModel>();

        }

    }
}

SplashActivity
    [Activity(Theme = "@style/MainTheme.Splash",
              MainLauncher = true,
              NoHistory = true,
              LaunchMode = LaunchMode.SingleTask)]
    public class SplashActivity : AppCompatActivity
    {
        // Launches the startup task
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            var data = Intent.GetStringExtra(LocalNotificationCenter.ReturnRequest);
            System.Diagnostics.Debug.WriteLine($"ReturnRequest data: {data}");
            var mainIntent = new Intent(Application.Context, typeof(MainActivity));
            System.Diagnostics.Debug.WriteLine(mainIntent);
            mainIntent.SetFlags(ActivityFlags.SingleTop);
            if (!string.IsNullOrWhiteSpace(data))
            {
                mainIntent.PutExtra(LocalNotificationCenter.ReturnRequest, data);
            }

            StartActivity(mainIntent);
        }
    }

MainActivity
    [Activity(Theme = "@style/MainTheme",
        LaunchMode = LaunchMode.SingleTask,
              ConfigurationChanges = ConfigChanges.ScreenSize | ConfigChanges.Orientation | ConfigChanges.UiMode | ConfigChanges.ScreenLayout | ConfigChanges.SmallestScreenSize)]
    public class MainActivity : global::Xamarin.Forms.Platform.Android.FormsAppCompatActivity
    {
        public const string NotificationKey = "NotificationKey";

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);

            Xamarin.Essentials.Platform.Init(this, savedInstanceState);

            global::Xamarin.Forms.Forms.Init(this, savedInstanceState);

            // Initialize NotificationCenter
            LocalNotificationCenter.CreateNotificationChannel(
                //channel 01 for night
                new NotificationChannelRequest
                {
                    Id = $"my_channel_01",
                    Name = "General",
                    Description = "General",
                    
                });

            LocalNotificationCenter.CreateNotificationChannel(
                    //channel 02 for morning
                    new NotificationChannelRequest
                    {
                        Id = $"my_channel_02",
                        Name = "Special",
                        Description = "Special",
                        
                    });
            LocalNotificationCenter.CreateNotificationChannel(
                     //channel 03 for weekly
                     new NotificationChannelRequest
                     {
                         Id = $"my_channel_03",
                         Name = "Weekly",
                         Description = "Weekly",
                        
                     });

            LoadApplication(new App(new AndroidInitializer()));
 
            LocalNotificationCenter.NotifyNotificationTapped(Intent);
        }

        protected override void OnNewIntent(Intent intent)
        {
            Console.WriteLine("OnNewIntent called");
            Intent = intent;
            string data = Intent.GetStringExtra(LocalNotificationCenter.ReturnRequest);
            Console.WriteLine($"ReturnRequest data: {data}");
            LocalNotificationCenter.NotifyNotificationTapped(Intent);
            base.OnNewIntent(intent);
           
           
        }

        public override void OnRequestPermissionsResult(int requestCode, string[] permissions, Android.Content.PM.Permission[] grantResults)
        {
            Xamarin.Essentials.Platform.OnRequestPermissionsResult(requestCode, permissions, grantResults);

            base.OnRequestPermissionsResult(requestCode, permissions, grantResults);
        }

        

    }

    public class AndroidInitializer : IPlatformInitializer
    {
        public void RegisterTypes(IContainerRegistry containerRegistry)
        {
            // Register any platform specific implementations
        }
    }

**Comment by thudugala** [comment]
@Cinmay2014 Can you please attach a sample project?

**Reply by Cinmay2014** [reply]
Thanks for replying me. I will attach a sample soon.

**Comment by Cinmay2014** [comment]
@thudugala sample project is here
https://github.com/Cinmay2014/sample01
I tried to use e.Request.NotificationId and e.Request.Android.ChannelId, but none of them works.
private async void Current_NotificationActionTapped(NotificationActionEventArgs e)
        {
            
            if (e.IsTapped)
            {
                Console.WriteLine($"e is {e.Request }");
                if (e.Request is null)
                {
                    return;
                }
                //if (e.Request.NotificationId == 88)
                if (e.Request.Android.ChannelId== "my_channel_01")
                {
                    
                    return;
                }
                else if (e.Request.Android.ChannelId== "my_channel_02")
                {
                    
                    await NavigationService.NavigateAsync("/Report");
                }
                else
                {
                   
                    await NavigationService.NavigateAsync("/WeeklyTips");
                }


            }
          
        }

**Reply by thudugala** [reply, marked as answer]
@Cinmay2014 

Also, need to set LocalNotificationCenter.ReturnRequestActionId

Doc updated

https://github.com/thudugala/Plugin.LocalNotification/wiki/Getting-NotificationTapped-to-work-with-splash-screen

**Reply by Cinmay2014** [reply]
Thanks @thudugala The tapped action works.
